src/components/NavBar.vue

```diff
@@ -0,0 +1,213 @@
+<template>
```

Comment:
nice abstraction of the navbar in its own vue component file.

Reply:
I am looking to optimize and split it more, but in the future